src/core/tests/MatrixViewUnitTest.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2015 Fondazione Istituto Italiano di Tecnologia
+ *
+ * Licensed under either the GNU Lesser General Public License v3.0 :
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ * or the GNU Lesser General Public License v2.1 :
+ * https://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * at your option.
+ */
+
+#include <iDynTree/Core/MatrixDynSize.h>
+#include <iDynTree/Core/MatrixView.h>
+#include <iDynTree/Core/TestUtils.h>
+#include <iDynTree/Core/EigenHelpers.h>
+
+using namespace iDynTree;
+
+template <class T, class U>
+void areMatricesEqual(const T & mat1, const U & mat2)
+{
+    ASSERT_EQUAL_DOUBLE(mat1.rows(), mat2.rows());
+    ASSERT_EQUAL_DOUBLE(mat1.cols(), mat2.cols());
+
+    ASSERT_EQUAL_MATRIX(mat1, mat2);
+}
+
+int main()
+{
+    // test with iDynTree matrix
+    MatrixDynSize mat1(27, 41);
+    getRandomMatrix(mat1);
+    MatrixView<double> view1(mat1);
+
+    areMatricesEqual(view1, mat1);
+
+    // Test with eigen column major matrix
+    Eigen::MatrixXd mat2(27, 41);
+    mat2.setRandom();
+    MatrixView<double> view2(mat2);
```

Comment:
How is the storage order detected here?

Reply:
An `Eigen::DenseBase<>` object has an attribute called `IsRowMajor`. Check [here]( https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#a86241c772c74c04eeeb0480b99c5ab77a406b6af91d61d348ba1c9764bdd66008). 

When you try to create a `MatrixView`, if `IsRowMajor` the following constructor is called
https://github.com/robotology/idyntree/blob/44694c92ac054b1e1a42d125ab78d8003addefff/src/core/include/iDynTree/Core/MatrixView.h#L121-L136

Reply:
Ah ok great, I was not able to find https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#a86241c772c74c04eeeb0480b99c5ab77a406b6af91d61d348ba1c9764bdd66008

I would add some tests on the storage orders anyway, to prevent future breakages.

Reply:
Done here: ba6bc38c5559ba6e31572cfc02d20355f84cba9b